### PR TITLE
Add create-or-update endpoint workflow; add EndpointResponseFuture

### PR DESCRIPTION
**Pull request by yixu34**
This PR has a couple of changes needed to support LSS:
* the `create_model_endpoint` function now takes a `update_if_exists`, which supports a create-or-update workflow. This is useful for continuing with our existing k8s-centric workflow, where we update deployments in place.
* Adds `get_model_endpoint` to support the above.
* Adds `EndpointResponseFuture` for easier polling of async tasks.

### CODE REVIEW

File: launch/client.py [hunk 1]
```diff
@@ -165,8 +166,9 @@ def create_model_bundle_from_dir(
         tmpdir = tempfile.mkdtemp()
         try:
             tmparchive = os.path.join(tmpdir, "bundle")
-            root_dir = os.path.dirname(base_path)
-            base_dir = os.path.basename(base_path)
+            abs_base_path = os.path.abspath(base_path)
```
Comment (yixu34) [inline launch/client.py:169]: This makes things more robust by actually expanding the absolute path.

File: launch/model_endpoint.py [hunk 2]
```diff
@@ -70,15 +71,45 @@ class EndpointResponse:
...
+class EndpointResponseFuture:
+    def __init__(self, client, async_task_id: str):
+        self.client = client
+        self.async_task_id = async_task_id
+
+    def get(self) -> EndpointResponse:
+        while True:
+            async_response = self.client.get_async_response(self.async_task_id)
+            if async_response["state"] == "PENDING":
+                time.sleep(2)
```
Comment (seanshi-scale) [inline launch/model_endpoint.py:93]: Should we do `async-await` stuff here? Or is that too complicated

File: launch/client.py [hunk 3]
```diff
@@ -492,6 +509,18 @@ def edit_model_endpoint(
...
 
+
+    def get_model_endpoint(self, endpoint_name: str) -> Optional[Union[AsyncEndpoint, SyncEndpoint]]:
+        try:
+            resp = self.connection.get(os.path.join(ENDPOINT_PATH, endpoint_name))
+        except APIError:
+            logger.exception("Got an error when retrieving endpoint %s", endpoint_name)
+            return None
+
+        # TODO: Return endpoint type and create the right type
```
Comment (seanshi-scale) [inline launch/client.py:520]: guessing this is still in progress? or does it require adding some stuff to the Gateway

File: launch/model_endpoint.py (hunk 2 above)
Comment (yixu34) [reply launch/model_endpoint.py:93]: I think this is fine for now. Looks like the Nucleus `sleep_until_complete` function does a blocking poll too.

IMO generally if we are doing async `await`, we're probably using an entirely `asyncio`-ified version of this client library, or at least a segment of the whole client (kinda like how `boto3` has `aioboto3`).

File: launch/client.py (hunk 3 above)
Comment (yixu34) [reply launch/client.py:520]: Oh yeah I think I'll just have the Gateway return the `endpoint_type`, will do that in the other PR.

Comment (seanshi-scale) [reply launch/client.py:543]: should we remove the TODO now?

**Review by seanshi-scale — APPROVED** [review]

**Pull request merged**